Allow expanding the diff if any changes are collapsed

hellcp-work opened the pull request:
Allows users to expand changes that weren't covered before

<img width="996" height="301" alt="Screenshot From 2025-08-20 16-48-03" src="https://github.com/user-attachments/assets/1e15e865-707e-4375-bdce-9e8d7b30d80a" />
<img width="996" height="301" alt="Screenshot From 2025-08-20 16-48-18" src="https://github.com/user-attachments/assets/f21f8e37-aba8-44a5-8bb1-f89d17c38531" />

rubhanazeem commented:
The current behavior is: if the diff is too big and not returned from the backend, we simply have the diff for such files disabled. Making an extra request doesn’t help, as it again returned empty diff content.

In this PR, I see that this behavior has been changed as expected, but the diff is still empty even after passing the tar limit. Why? Shouldn’t we be getting a diff? I think in these cases the accordion should stay disabled so the user can’t click and expect a diff. Have a look at the backend response I get after passing the tar limit.

```
 "filenames" => ["vlogger-1.3.tar.gz"],
 "files" =>
  {"vlogger-1.3.tar.gz" =>
    {"state" => "added", "new" => {"name" => "vlogger-1.3.tar.gz", "md5" => "4170a4bf7ab8b24373458e8ac820c0a1", "size" => "17715"}, "diff" => {"binary" => "1", "lines" => "0"}}},
 "issues" => {}}
```

<img width="1477" height="77" alt="image" src="https://github.com/user-attachments/assets/a3469a2c-9fa5-4390-8b33-63f2dec06401" />

hellcp-work commented:
> ```
>  "filenames" => ["vlogger-1.3.tar.gz"],
>  "files" =>
>   {"vlogger-1.3.tar.gz" =>
>     {"state" => "added", "new" => {"name" => "vlogger-1.3.tar.gz", "md5" => "4170a4bf7ab8b24373458e8ac820c0a1", "size" => "17715"}, "diff" => {"binary" => "1", "lines" => "0"}}},
>  "issues" => {}}
> ```

What you are showing off is an issue with added files, this PR only addresses issues with changed files. I thought there is another card for added and deleted diffs, since it requires backend changes.

hellcp-work commented:
The expansion is now only proposed for changed files

rubhanazeem commented on src/api/app/components/sourcediff_component.rb:
```diff
@@ -42,12 +54,12 @@ def target_package
 
-    files.each_with_index do |(filename, contents), file_index|
-      files[filename]['disabled'] = contents['diff'].nil? || contents['diff']['_content'].nil?
-      next if files[filename]['disabled']
```
Should we keep it for added files where diff content is nil?

hennevogel commented on src/api/app/components/sourcediff_component.html.haml:
```diff
@@ -7,8 +7,14 @@
+        - if max_tar_limit.to_i >= BsRequestAction::Differ::ForSource::DEFAULT_FILE_LIMIT && max_tar_limit.to_i >= tarlimit.to_i
+          .alert.alert-warning
+            Some of the changes are collapsed due to size.
+            = link_to(request_changes_path(@bs_request, @action, tarlimit: max_tar_limit)) do
```
Why do you calculate `max_tar_limit`? Isn't there something like "no limit"?

hellcp-work replied:
Apparently there is, setting tarlimit to 0, and I missed it

hellcp-work commented on src/api/app/components/sourcediff_component.rb:
```diff
@@ -1,17 +1,30 @@
+  def expandable?(sourcediff)
+    return false unless sourcediff.key?('files')
+
+    sourcediff['files'].filter_map do |f, d|
```
Done

danidoni commented on src/api/app/jobs/bs_request_action_webui_infos_job.rb:
```diff
@@ -53,4 +54,14 @@ def for_a_superseded_request(superseded_request, bs_request_action, source_packa
   end
+
+  def for_tar_file_limit(bs_request_action, source_package_names, tarlimit)
+    silent do
```
What this `silent` block does?

danidoni commented on src/api/app/views/webui/request/_changes_maintenance_incident.html.haml:
```diff
@@ -9,4 +9,5 @@
 
-= render SourcediffComponent.new(bs_request: bs_request, action: action, diff_to_superseded: diff_to_superseded, diff_not_cached: diff_not_cached)
+= render SourcediffComponent.new(bs_request: bs_request, action: action, diff_to_superseded: diff_to_superseded,
+                                 diff_not_cached: diff_not_cached, tarlimit: tarlimit, tarlimit: tarlimit)
```
```suggestion
                                 diff_not_cached: diff_not_cached, tarlimit: tarlimit)
```

danidoni replied:
There is an extra `tarlimit` parameter
